### ISSUE TITLE
[RMP] - Incorporate embeddings (pre-trained and encoder-based) into Transformers4Rec Training and Inference

### ISSUE THREAD

**Issue by MarkMoTrin** [NVIDIA-Merlin/Merlin#471]
## Problem:
Two+ teams at Walmart (interest from Lowes Recsys team as well) would like to leverage encoder-based embeddings as additional features in Transformers4Rec. Currently, there is no clear example/path to pass in extra embeddings. This gap in feature prevents teams from adopting Transformers4Rec in production.

## Goal:
- Produce a comprehensive example on how to incorporate embeddings as extra features into Transformers4Rec. Embeddings may be stored in parquet format. Generalizations should be made where possible in examples, especially on defining the schema and other input parameters.

## Constraints:
- Embeddings may be stored in parquet files
- Workflow should be able to be deployed in Triton

## Starting Point:
[ ] Write a short blog post describing how embeddings can be included in Transformers4Rec framework (theoretically), show embedding sizes as data gets normalized and sent to T4Rec
[ ] Highlight the needed input parameters to pass in embedding type data into the schema object, NVTabular, and T4Rec.

It is _critical_ that this ticket not only be created, but also kept up to date.  As you work constraints are going to be discovered and should be added to the above list.  Tasks required to complete this project may change.  The goal of the work may even change.  Without a commitment to keeping this ticket up to date the work shouldn't be undertaken.  

Example tickets from which you can base your own roadmap issue include:
- https://github.com/NVIDIA-Merlin/models/issues/450
- https://github.com/NVIDIA-Merlin/Merlin/issues/258
- https://github.com/NVIDIA-Merlin/Merlin/issues/279

**Comment by MarkMoTrin** [comment]
[using-model-embeddings-nvtabular.pdf](https://github.com/NVIDIA-Merlin/Merlin/files/9168775/using-model-embeddings-nvtabular.pdf)

**Comment by viswa-nvidia** [comment]
Removing the milestone 22.09. This is not in the POR yet.

**Comment by viswa-nvidia** [comment]
@benfred , @jperez999 , 
Review this ticket and add planning details.  It is still in needs definition.
Convert the check boxes to tickets in the specific repos ( Please remember to update the initiative field )
Once you are satisfied with the planning change status to RMP review
[Refer to this doc for the workflow](https://docs.google.com/document/d/1tyq9sWZvgF3nO3R4r3knOoXAXRjDhNAq0qRsqHTdb6o/edit#heading=h.kdc61ua1sb25)
[There is a link to the video](https://docs.google.com/document/d/1tyq9sWZvgF3nO3R4r3knOoXAXRjDhNAq0qRsqHTdb6o/edit#heading=h.gxey53iplz6) that explains the workflow. Let me know if you have any questions.

**Comment by oliverholworthy** [comment]
For small embedding tables (fit in GPU memory). Using a tensor initializer is the currently supported functionality for pre-trained embedding tables. An example in [this notebook](https://github.com/NVIDIA-Merlin/models/blob/1dfea5ef6196468420749bf9a7fbfaeef838f20c/examples/usecases/entertainment-with-pretrained-embeddings.ipynb ):

**Comment by karlhigley** [comment]
@MarkMoTrin When you say "encoder-based embeddings as additional features," does that imply encoding an item attribute (like "text description") using an external model (like BERT) and supplying the resulting embeddings as features for each item in a session? We're not talking about adding a single embedding as a session-level feature, we're talking about adding a list of embeddings as item-level features, yeah?

**Comment by gabrielspmoreira** [comment]
When the embedding table are not huge and fit GPU memory, the new `PretrainedEmbeddingsInitializer` ( https://github.com/NVIDIA-Merlin/Transformers4Rec/pull/572 ) can be used to initialize the embedding matrix with pre-trained embeddings and set them to trainable or not.
